### DISCUSSION TITLE
Resolving dependencies in workspaces (double-bundling React)

### DISCUSSION THREAD

**Discussion by kukuruznik** [parcel-bundler/parcel#8510] [Q&A] [answered]
Hello,

I'm working on a React project along with a React component library. Because of the concurrent development, the library is referenced in the project as an npm workspace.

The library has a dev (and peer)  dependency on React. When the library is packaged and installed in the project (and not used as a workspace), things work fine. But when the library is linked as a workspace dependency, Parcel (probably) packages the React dependency of the library alongside with the React dependency of the project and the front-end fails with "Invalid hook call. Hooks can only be called inside of the body of a function component." and "TypeError: dispatcher is null".

Is there a known method, how to use Parcel with workspaces correctly? Could you please point me in the right direction? Googling has been futile so far...

Thank you in advance!

**Comment by mischnic** [comment, marked as answer]
There's no way for Parcel to correctly deduplicate React in this case.

I'd say the usual workflow is either using a monorepo (where npm/Yarn take care of deduplication), or not linking and instead using `file:../my-component` (or something like that).

A hacky workaround: [alias `react` to a global variable](https://parceljs.org/features/dependency-resolution/#global-aliases) and then in your project assign that global variable (so that there is effectively only one React import).
